pkg/api/v1beta1/types.go

```diff
@@ -1195,3 +1195,92 @@ type SecretList struct {
 
 	Items []Secret `json:"items" description:"items is a list of secret objects"`
 }
+
+// AutoScaler monitors other resources that are resizeable and adjusts them according to configuration.
+type AutoScaler struct {
+	TypeMeta `json:",inline"`
+
+	// Spec defines the auto-scaler targets and thresholds.
+	Spec AutoScalerSpec `json:"spec,omitempty"`
+
+	// Status defines the actions the auto-scaler has taken.
+	Status AutoScalerStatus `json:"status,omitempty"`
+}
+
+// AutoScalerStatus provides the status of an auto-scaler.
+type AutoScalerStatus struct {
+	// TODO: open for discussion on what meaningful information can be reported in the status
+	// The status may return the replica count here but we may want more information
+	// such as if the count reflects a threshold being passed.
+}
+
+// AutoScalerSpec defines the auto-scaler targets and thresholds.
+type AutoScalerSpec struct {
+	// Thresholds holds a collection of AutoScaleThresholds that drive the auto-scaler.
+	Thresholds []AutoScaleThreshold `json:"thresholds,omitempty"`
+
+	// Enabled turns auto scaling on or off.
+	Enabled bool `json:"enabled,omitempty"`
+
+	// MaxAutoScaleCount defines the max replicas that the auto-scaler can use.  This value must be
+	// >= MinAutoScaleCount.
+	MaxAutoScaleCount int `json:"maxAutoScaleCount,omitempty"`
+
+	// MinAutoScaleCount defines the minimum number replicas that the auto-scaler can reduce to,
+	// 0 means that the application is allowed to idle.
+	MinAutoScaleCount int `json:"minAutoScaleCount,omitempty"`
+
+	// TargetSelector provides the resizeable target(s).  Right now this is a ReplicationController
+	// in the future it could be a job or any resource that implements resize.  If multiple targets
+	// are resolved by the selector the auto-scaler will resize the largest one.
+	TargetSelector map[string]string `json:"targetSelector,omitempty"`
```

Comment:
Sync with Dario


> On Mar 16, 2015, at 10:18 AM, Paul <notifications@github.com> wrote:
> 
> In pkg/api/v1beta1/types.go:
> 
> > +
> > +	// Enabled turns auto scaling on or off.
> > +	Enabled bool `json:"enabled,omitempty"`
> > +
> > +	// MaxAutoScaleCount defines the max replicas that the auto-scaler can use.  This value must be
> > +	// >= MinAutoScaleCount.
> > +	MaxAutoScaleCount int `json:"maxAutoScaleCount,omitempty"`
> > +
> > +	// MinAutoScaleCount defines the minimum number replicas that the auto-scaler can reduce to,
> > +	// 0 means that the application is allowed to idle.
> > +	MinAutoScaleCount int `json:"minAutoScaleCount,omitempty"`
> > +
> > +	// TargetSelector provides the resizeable target(s).  Right now this is a ReplicationController
> > +	// in the future it could be a job or any resource that implements resize.  If multiple targets
> > +	// are resolved by the selector the auto-scaler will resize the largest one.
> > +	TargetSelector map[string]string `json:"targetSelector,omitempty"`
> To be clear, are you asking me to change this into a labels.Set? It looks like everything is still using maps of strings in the type definitions.
> 
> —
> Reply to this email directly or view it on GitHub.
>